packages/web-wallet/src/components/WalletHome.tsx

```diff
@@ -0,0 +1,312 @@
+                  }`}
+              >
+                <ArrowUpRight className="w-4 h-4 text-white" />
+                <span className="text-sm font-medium text-white">Send</span>
+              </button>
+              <button
+                onClick={() => handleButtonClick('Receive')}
+                className={`flex-1 md:flex-none px-6 py-2.5 bg-primary hover:bg-primary/90 rounded-xl flex items-center justify-center gap-2 transition-colors ${activeButton === 'Receive' ? 'bg-primary/90' : ''
+                  }`}
+              >
+                <ArrowDownLeft className="w-4 h-4 text-white" />
+                <span className="text-sm font-medium text-white">Receive</span>
+              </button>
+            </div>
+          </div>
+        </div>
+
+        {/* My Assets Section */}
+        <div className="space-y-6">
+          {/* Section Header */}
+          <h2 className="text-xl font-medium text-white">My Assets</h2>
+
+          {/* Token Tabs */}
+          <TokenTabs
+            selectedFilter={selectedTokenFilter}
+            onFilterChange={(filter) => updateSearchParams({ filter })}
+            tokenCount={tokenCount}
+            nftCount={nftCount}
+          />
+
+          {/* Token List */}
+          <TokenList
+            tokens={tokens}
+            onTokenClick={(tokenUid) => {
+              navigate(`?dialog=history&token=${tokenUid}`);
+            }}
+            onSendClick={(tokenUid) => {
+              navigate(`?dialog=send&token=${tokenUid}`);
+            }}
+          />
+        </div>
+      </div>
+
+      {/* Dialogs */}
+      <SendDialog
+        isOpen={sendDialogOpen}
+        onClose={closeDialog}
+        initialTokenUid={tokenUid || undefined}
+      />
+      <ReceiveDialog
+        isOpen={receiveDialogOpen}
+        onClose={closeDialog}
+      />
+      <HistoryDialog
+        isOpen={historyDialogOpen}
+        onClose={closeDialog}
+        tokenUid={tokenUid || undefined}
+        onRegisterTokenClick={() => navigate('?dialog=register')}
+      />
+      <RegisterTokenDialog
+        isOpen={registerTokenOpen}
+        onClose={closeDialog}
+      />
+      <CreateTokenDialog
+        isOpen={createTokenOpen}
+        onClose={closeDialog}
+      />
+      <AddressModeDialog
+        isOpen={addressModeOpen}
+        onClose={closeDialog}
+      />
+
+      {/* Error notification */}
+      {error && (
+        <ErrorNotification
+          error={new Error(error)}
+          onDismiss={() => setError(null)}
+        />
+      )}
+
+      {/* Transaction notification */}
+      {(() => {
+        if (!newTransaction) return null;
+        const tx = newTransaction as Record<string, unknown>;
+        if (tx.tx_id) return null; // This is for history dialog, not notification
+
+        const notification = tx as { type: 'sent' | 'received'; amount: bigint; timestamp: number };
+        return (
+          <TransactionNotification
```

Comment:
What happens if we receive 2 transactions close to each other?

Reply:
I commented out transaction notification for now